openstates/pr/legislators.py

```diff
@@ -57,100 +58,113 @@ def scrape_senate(self, term):
-                    leg.add_office('district', 'Oficina de Distrito',
-                                   address=district_addr,
-                                   phone=district_phone,
-                                   fax=district_fax)
-
-                leg.add_source(url)
-                self.save_legislator(leg)
+                else:
+                    self.warning(u"{} represents a district, but it is not listed".
+                            format(name))
+                    missing_district_on_index_page = True
+
+                    # Special-case one member, whose page only displays a SQL error
+                    if name == u"Yashira M. Lebrón Rodríguez":
+                        leg = Legislator(
+                                term=term,
+                                chamber='lower',
+                                district='8',
+                                full_name=name,
+                                party=party
+                                )
+                        leg.add_source(url)
+                        self.save_legislator(leg)
+
+                        special_case_still_needed = True
+                        continue
+
+            # Parse the member's webpage for contact information
+            member_doc = self.lxmlize(member_url)
+
+            (email, ) = member_doc.xpath('//a[starts-with(@href, "mailto:")]/text()')
+            phone_and_fax = [
+                    x.strip() for x in
+                    member_doc.xpath(u'//b[contains(text(), "Teléfono(s):")]/..//text()')
+                    if x.strip()
+                    ]
+            try:
+                phone_index = phone_and_fax.index(u"Teléfono(s):") + 1
+                phone = phone_and_fax[phone_index]
+            except IndexError:
+                phone = None
+            try:
+                fax_index = phone_and_fax.index(u"Fax:") + 1
+                fax = phone_and_fax[fax_index]
+            except ValueError:
+                fax = None
+
+            if missing_district_on_index_page:
+                (district, ) = member_doc.xpath('//div[@class="tbrown"]/span/b/text()')
+                district = re.search(r'Distrito 0?(\d{1,2})', district).group(1)
+
+            leg = Legislator(
+                    term=term,
+                    chamber='lower',
+                    district=district,
+                    full_name=name,
+                    party=party,
+                    photo_url=photo_url
+                    )
+
+            leg.add_source(url)
+            leg.add_source(member_url)
+            leg.add_office(
+                    type='capitol',
+                    name='Oficina del Capitolio',
+                    phone=phone,
+                    fax=fax,
+                    email=email
+                    )
+
+            self.save_legislator(leg)
+
+        # If this `assert` breaks, then remove the special-casing code
+        assert special_case_still_needed
```

Comment:
We should change this to be:

`assert special_case_still_needed, "Yashira M. Lebrón Rodríguez doesn't need to be special cased anymore"` so we can scan the tracebacks